var/spack/repos/builtin/packages/mxnet/package.py

```diff
@@ -0,0 +1,101 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Mxnet(MakefilePackage):
+    """MXNet is a deep learning framework
+    designed for both efficiency and flexibility."""
+
+    homepage = "http://mxnet.io"
+    url      = "https://github.com/apache/incubator-mxnet/archive/0.10.0.post2.tar.gz"
+
+    version('0.10.0.post2',  '7819d511cf4a6efad681e6662fa966e4')
+    version('0.10.0.post1',  '16d540f407cd22285555b3ab22040032',
+       	    url="https://github.com/apache/incubator-mxnet/archive/v0.10.0.post1.tar.gz")
+    version('0.10.0', '2d0c83c33eda729932d620cca3078826',
+       	    url="https://github.com/apache/incubator-mxnet/archive/v0.10.0.tar.gz")
+
+    variant('cuda', default=False, description='Enable CUDA support')
+    variant('opencv', default=True, description='Enable OpenCV support')
+
+    depends_on('dmlc-core')
+    depends_on('mshadow')
+    depends_on('ps-lite')
+    depends_on('nnvm')
+    depends_on('openblas')
+    depends_on('cudnn', when='+cuda')
+    depends_on('cudnn', when='+cuda')
+    depends_on('cub', when='+cuda')
+    depends_on('opencv+imgproc+highgui+jpeg+png+tiff~eigen~ipp@3.0:', when='+opencv')
+
+    # Lock version for the stable release 0.10.0
+    depends_on('dmlc-core@a6c5701', when='@0.10.0')
+    depends_on('mshadow@c037b06', when='@0.10.0')
+    depends_on('ps-lite@acdb698', when='@0.10.0')
+    depends_on('nnvm@b279286', when='@0.10.0')
+
+    patch('makefile.patch', when='@0.10:0.11')
+
+    def build(self, spec, prefix):
+	filter_file('export CC = gcc', '', 'make/config.mk', string=True)
+	filter_file('export CXX = g++', '', 'make/config.mk', string=True)
```

Comment:
Moving `filter_file` into `edit()` leads to the following error:

```
==> Error: TypeError: edit() takes exactly 1 argument (3 given)                                                    
/lustre/home/acct-hpc/hpc-jianwen/spack/lib/spack/spack/package.py:1350, in build_process:                       
     1278         def build_process(input_stream):                                                                 
     1279             """Forked for each build. Has its own process and python                                     
     1280                module space set up by build_environment.fork()."""       
...
     1349                 # Run post install hooks before build stage is removed.
  >> 1350                 spack.hooks.post_install(self.spec)
     1351
     1352             # Stop timer.
...
```

Code in `mxnet/package.py`.
```
def edit(self):
    filter_file('export CC = gcc', '', 'make/config.mk', string=True)
    filter_file('export CXX = g++', '', 'make/config.mk', string=True)
```